com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -39,23 +41,136 @@
 
   @Getter private List<SyntaxError> errors = new ArrayList<>();
 
+  private String programName = null;
   private String documentUri;
   private SemanticContext semanticContext;
+  private CommonTokenStream tokenStream;
 
-  public CobolVisitor(String documentUri, SemanticContext semanticContext) {
+  public CobolVisitor(
+      String documentUri, SemanticContext semanticContext, CommonTokenStream tokenStream) {
     this.documentUri = documentUri;
     this.semanticContext = semanticContext;
+    this.tokenStream = tokenStream;
+  }
+
+  @Override
+  public Class visitIdentificationDivision(IdentificationDivisionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitProgramIdParagraph(ProgramIdParagraphContext ctx) {
+    if (ctx.programName() != null) {
+      programName = ctx.programName().getText();
+    }
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitProcedureDivision(ProcedureDivisionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitEnvironmentDivision(EnvironmentDivisionContext ctx) {
+    Token token = ctx.getStart();
+    areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    return visitChildren(ctx);
```

Comment:
Question: May this portion of code be extracted in a private method that accept as input an instance of ```ParserRuleContext```?